lib/segment/src/segment.rs

```diff
@@ -325,14 +340,22 @@ impl Segment {
         let id_tracker = self.id_tracker.borrow();
         internal_result
             .iter()
-            .map(|&scored_point_offset| {
+            .filter_map(|&scored_point_offset| {
```

Comment:
Theoretically, it is possible to get such a scenario:
The user searches for the nearest point and we filter here this point. The user becomes an empty success response without any points.

Reply:
possible

Reply:
Is it okay? Can we research with more neighbors if we got unexists points?

Reply:
it is not okay, but better than panic